src/eko/output/manipulate.py

```diff
@@ -175,5 +186,9 @@ def to_evol(eko: EKO, source: bool = True, target: bool = False):
     # assign pids
     if source:
         eko.rotations._inputpids = br.evol_basis_pids
+        # update metadata
+        eko.update_metadata({"rotations": {"inputpids": inputpids}})
```

Comment:
At this point I believe that in any case `rotations.inputpids` and `rotations.targetpids` should be matrices, not only for `to_evol`. The question is, do we want to store the matrix itself inside the file `metadata` or do we want to store the vector and then internally change the object (for example `rotations.targetpids`)? In the second case we can add something in `__post__init__`: Since we have both the vectors `rotations.pids` and `rotations.targetpids` as readed from `metadata` we can just compute the matrix and assign it to `rotations.targetgrid` in place of the vector. In this way, we will only store vectors, which I believe are more clear to read for an user, but then internally we have the matrices we need. Do you agree?

Reply:
As we discussed yesterday, we want the matrix: it is uniform, also for non-standard bases, and always reliable.
The less special cases we have, the easier will be the code, and maintainability will improve.